fgspectra/frequency.py

```diff
@@ -124,64 +131,92 @@ class ThermalSZ(SED):
+
+    def __call__(self, nu, *args):
+        # Return the evaluation of the SED
+        return np.ones(np.array(nu).shape)
 
-r'''
 class CIB(SED):
-    r"""
-    SED for :math:`g(\nu)` for CIB models, where
-
-    .. math:: g(\mu) = (\partial B_{\nu}/\partial T)^{-1} |_{T_CMB}
-
-    Parameters
-    ----------
-    nu: float or array
-        Frequency in Hz.
-    beta: float
-        Spectral index.
-
-    Methods
-    -------
-    __call__(self, nu, beta)
-        return the frequency scaling given by a power law.
-    """
-    def g(self, nu, T_CMB):
-        """Convert from flux to thermodynamic units."""
-        x = constants.h * (nu*1e9) / (constants.k * T_CMB)
-        return constants.c**2 * constants.k * T_CMB**2 * (np.cosh(x) - 1) / (
-            constants.h**2 * (nu*1e9)**4)
 
-    def planckB(self, nu, T_d):
+
+    @staticmethod
+    def planckB(nu, T_d):
         """Planck function at dust temperature."""
         x = constants.h * (nu*1e9) / (constants.k * T_d)
         return 2 * constants.h * (nu*1e9)**3 / constants.c**2 / (np.exp(x) - 1)
 
-    def sed(self, nu, beta, T_d, T_CMB):
-        """Modified blackbody mu(nu, beta)"""
-        return nu**beta * self.planckB(nu, T_d) * self.g(nu, T_CMB)
-'''
+    @staticmethod
+    def mu(nu, beta, T_d, T_CMB):
+        return nu**beta * CIB.planckB(nu, T_d) * g(nu, T_CMB)
+
+    def __call__(self, nu, beta, T_d, nu_0, T_CMB=2.725):
+        """Compute the SED with the given frequency and parameters.
+
+        nu : float
+            Frequency in GHz.
+        beta : power law parameter
+        T_d : dust temperature
+        T_CMB (optional) : float
+        """
+        return CIB.mu(nu, beta, T_d, T_CMB) / CIB.mu(nu_0, beta, T_d, T_CMB)
+
+
+class tSZxCIB(SED):
+    def __init__(self, effective_frequencies=False):
+        """This is the only object for which it matters to have legacy
+        support for effective frequencies."""
+        self.effective_frequencies = effective_frequencies
+
+    def __call__(self, nu_1, nu_2, beta, nu_0, T_d=9.7, T_CMB=2.725):
+        """Compute the SED with the given frequency and parameters.
+
+        nu : float
+            Frequency in GHz.
+        beta : power law parameter
+        T_d : dust temperature
+        T_CMB (optional) : float
+        """
+        beta = np.array(beta)[..., np.newaxis]
+        T_d = np.array(T_d)[..., np.newaxis]
+        if self.effective_frequencies:
+            return np.sqrt( ThermalSZ.f(nu_1, T_CMB)[...,np.newaxis] * CIB.mu(nu_2, beta, T_d, T_CMB)
+                         / (ThermalSZ.f(nu_0, T_CMB) * CIB.mu(nu_0, beta, T_d, T_CMB)))
+        else:
+            return ( (ThermalSZ.f(nu_1, T_CMB)[...,np.newaxis] * CIB.mu(nu_2, beta, T_d, T_CMB) +
+                         CIB.mu(nu_1, beta, T_d, T_CMB)[...,np.newaxis] * ThermalSZ.f(nu_2, T_CMB))
+                         / (2 * ThermalSZ.f(nu_0, T_CMB) * CIB.mu(nu_0, beta, T_d, T_CMB)))
 
 
-# CMB
-# blackbody, derivative BB
+class PowerLaw_g(SED):
```

Comment:
Is this just the standard PL with different units? We should probably make a decision about how to handle them